### ISSUE TITLE
Slack Login Authentication function

### ISSUE THREAD

**Issue by m4ndycheung** [m4ndycheung/cyf-stay-tuned-radio#46]
Dependent on tickets #45 and #47 to be completed so we have a CLIENT ID and CLIENT SECRET.

To do:
- [ ]  Get part 1 and 2 working in this diagram: https://api.slack.com/authentication/oauth-v2#obtaining

Dev Notes:
- https://slack.dev/node-slack-sdk/oauth (Read limitations)
- https://slack.dev/node-slack-sdk/web-api#sign-in-with-slack-via-openid-connect
- https://github.com/slackapi/node-slack-sdk/tree/main/examples/openid-connect
- ngrok

**Comment by m4ndycheung** [comment]
Hello teamies :) I had a look into getting the Slack Access Tokens and ran into some issues along the way. 
If you run into any of these, the following may help you:
- The [Slack github](https://github.com/slackapi/node-slack-sdk/blob/main/examples/openid-connect/app.js) is very useful but I struggled with these things:
- We've already created an app, the instructions talk about creating one with a Manifest - that's basically a blueprint for the app configuration.
- ngrok - Douglas said we may not need to use this but we do if we want to test the Slack App on localhost. As in the Slack app, when changing the redirect url, it won't accept anything that starts with "http", only "https".
- I found the instructions in the [Slack github](https://github.com/slackapi/node-slack-sdk/blob/main/examples/openid-connect/app.js) to be on the sparse side, so I signed up to ngrok and followed the instructions to install it here: https://ngrok.com/docs/getting-started/?os=linux
- If you're using ngrok, you'll need to "start it up". It looks like this:

![Image](https://github.com/m4ndycheung/cyf-stay-tuned-radio/assets/109825538/3c35daf5-17ff-4a5b-b516-c0d8846d268e)
-Everytime you shut down ngrok in terminal and restart it, your redirect url will also change, and it'll need to be changed in the app and .env 

- The slack Github is also written in Kao and Router instead of Express
